File: 1603-design-parking-system/1603-design-parking-system.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        # Store available slots using a dictionary indexed by carType (1, 2, 3)
        self.slots = {
            1: big,
            2: medium,
            3: small
        }

    def addCar(self, carType: int) -> bool:
        # Check if there are available slots for the given carType
        if self.slots[carType] > 0:
            self.slots[carType] -= 1
            return True
        return False


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

File: 1603-design-parking-system/1603-design-parking-system.java
class ParkingSystem {
    private int[] slots;

    public ParkingSystem(int big, int medium, int small) {
        // Index 0 unused, index 1 = big, 2 = medium, 3 = small
        this.slots = new int[]{0, big, medium, small};
    }
    
    public boolean addCar(int carType) {
        if (this.slots[carType] > 0) {
            this.slots[carType]--;
            return true;
        }
        return false;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */